### PR TITLE
Add basic support for converting Avro records to JSON for output

### PR DESCRIPTION
**Pull request by llofberg**
This change is [Reviewable](https://reviewable.io/reviews/edenhill/kafkacat/151)

### CODE REVIEW

**Comment by Marcelovk** [comment]
This is awesome, I was able to compile with it and is much better than using kafka-avro-console-consumer. The only thing that I missed is the documentation in the command line describing -a,-A and -s options

File: configure.kafkacat [hunk 1]
```diff
@@ -43,8 +43,22 @@ struct rd_kafka_metadata foo;"
...
     fi
+
+
+
+    mkl_meta_set "libserdes" "deb" "libserdes-dev"
+    # Check for Avro serdes library (serdes)
+    if [[ $WITH_AVRO == y ]] && \
+        mkl_lib_check --static=-lserdes "serdes" HAVE_SERDES disable CC "-lserdes -lavro" \
+        "#include  <stdlib.h>
+        #include  <libserdes/serdes-avro.h>
```
Comment (sknop) [inline configure.kafkacat]: It looks like on Mac you need to include 
#include <sys/types.h> 
as well or the test compile will fail.

**Comment by ryansobol** [comment]
Patiently awaiting this feature...

**Comment by homer6** [comment]
@edenhill @sknop What is still required for this to be merged into the mainline?

**Comment by edenhill** [comment]
Sorry for the delay, busy finalizing the librdkafka v1.0 release. Will tend to this in a week or two.

**Comment by arithehun** [comment]
@casidiablo You also have to install avro-c to compile:

```brew install avro-c```

**Comment by KarthikRangaraju** [comment]
Looking forward to this feature to get merged 👍

**Comment by arithehun** [comment]
@edenhill What's the ETA on getting this merged?

**Comment by edenhill** [comment]
librdkafka v1.0.0 is going out tonight, should be able to address kafkacat PRs next week and get a new kafkacat release out.

**Comment by jataggart** [comment]
this would be really great to have, any updates?

**Comment by bigkraig** [comment]
friendly ping @edenhill

**Comment by kyleholzinger** [comment]
Any update on this? I use kafkacat quite often and would love this support!

**Comment by arithehun** [comment]
@edenhill There was a lot of work that went into this patch and it contains a critical feature for streaming kafka records at any company that uses avro. It is a major blocker and it has deterred many from using kafkacat. This change is ready to merge in to master - myself and several colleagues have built it locally and it works flawlessly

Please let us know if you still intend on maintaining kafkacat and if we should instead invest in forking this repository and setting up our own brew mirroring host so that we can continue to sustain contributions from the open source community.

**Comment by arithehun** [comment]
I understand if you are swamped with work and might also consider accepting other people to maintain this repo. I appreciate all the hard work you've put into this tool and I'm sure we'd love to help continue support your workload for continuing on adding new features

**Comment by edenhill** [comment]
Please try out this docker image to consume Avro-encoded messages:
`docker run edenhill/kafkacat:sr-wip -b YOURBROKER -s YOUR_SR_URL -C -t YOUR_TOPIC -a value -Ee -J | jq .payload.some.avro.field`

**Comment by cecchisandrone** [comment]
> Please try out this docker image to consume Avro-encoded messages:
> `docker run edenhill/kafkacat:sr-wip -b YOURBROKER -s YOUR_SR_URL -C -t YOUR_TOPIC -a value -Ee -J | jq .payload.some.avro.field`

I tried and I receive `"payload_error": "Invalid CP1 magic byte 32, expected 0"`. there is no request to schema registry. I was wondering what is the subject convention to gather the schema. Is it modifiable?

**Comment by edenhill** [comment]
@cecchisandrone That means the message value/payload was not serialized by a schema-registry-aware producer.

The schema-registry-aware avro producer prefixes the serialized avro data with a short header that includes a magic value and the schema-id, this is what the consumer uses to look up the schema and deserialize the message.

**Comment by Marcelovk** [comment]
@edenhill That command works correcly for me. I also tried with `-f "%k : %s\n"` . This feature will be a great addition to kafkacat

**Comment by cecchisandrone** [comment]
Tested with a correct topic. It seems to work.

**Comment by edenhill** [comment]
Added generic deserializers (for various ints, char, string, inpspired by Python's struct.unpack) and changed the command line arguments.
I believe this is a more flexible and future-proof path forward.

Please try out this latest change in with `edenhill/kafkacat:serdes-wip`  (docker image).

**Comment by edenhill** [comment]
Thanks for your contribution and your patience!

**Comment by cecchisandrone** [comment]
Is there also a plan to support for AVRO serialization?

**Comment by cecchisandrone** [comment]
Also, I wanted to test this on latest master build on windows. Should I need to enable something in the build config? I receive this message after building kafkacat on windows:
`% ERROR: This build of kafkacat lacks Avro/Schema-Registry support`

**Comment by edenhill** [comment]
We'll add Avro serialization in a later version (incremental delivery, aight?!).

Windows support is further down the line since libserdes is not supported on Windows yet.

**Pull request merged**